RFC-0001-prsestissimo-metrics.md

```diff
@@ -0,0 +1,251 @@
+velox_hive_file_handle_generate_latency_ms_bucket{cluster="testing",worker="",le="70000"} 0
+velox_hive_file_handle_generate_latency_ms_bucket{cluster="testing",worker="",le="80000"} 0
+velox_hive_file_handle_generate_latency_ms_bucket{cluster="testing",worker="",le="90000"} 0
+velox_hive_file_handle_generate_latency_ms_bucket{cluster="testing",worker="",le="100000"} 0
+velox_hive_file_handle_generate_latency_ms_bucket{cluster="testing",worker="",le="+Inf"} 0
+```
+
+Summaries
+```
+# TYPE presto_cpp_http_client_presto_exchange_source_on_body_bytes_summary summary
+presto_cpp_http_client_presto_exchange_source_on_body_bytes_summary_count{cluster="testing",worker=""} 0
+presto_cpp_http_client_presto_exchange_source_on_body_bytes_summary_sum{cluster="testing",worker=""} 0
+presto_cpp_http_client_presto_exchange_source_on_body_bytes_summary{cluster="testing",worker="",quantile="0.5"} Nan
+presto_cpp_http_client_presto_exchange_source_on_body_bytes_summary{cluster="testing",worker="",quantile="0.9"} Nan
+presto_cpp_http_client_presto_exchange_source_on_body_bytes_summary{cluster="testing",worker="",quantile="0.95"} Nan
+presto_cpp_http_client_presto_exchange_source_on_body_bytes_summary{cluster="testing",worker="",quantile="0.99"} Nan
+presto_cpp_http_client_presto_exchange_source_on_body_bytes_summary{cluster="testing",worker="",quantile="1"} Nan
+# TYPE presto_cpp_presto_exchange_source_serialized_page_size_summary summary
+presto_cpp_presto_exchange_source_serialized_page_size_summary_count{cluster="testing",worker=""} 0
+presto_cpp_presto_exchange_source_serialized_page_size_summary_sum{cluster="testing",worker=""} 0
+presto_cpp_presto_exchange_source_serialized_page_size_summary{cluster="testing",worker="",quantile="0.5"} Nan
+presto_cpp_presto_exchange_source_serialized_page_size_summary{cluster="testing",worker="",quantile="0.9"} Nan
+presto_cpp_presto_exchange_source_serialized_page_size_summary{cluster="testing",worker="",quantile="0.95"} Nan
+presto_cpp_presto_exchange_source_serialized_page_size_summary{cluster="testing",worker="",quantile="0.99"} Nan
+presto_cpp_presto_exchange_source_serialized_page_size_summary{cluster="testing",worker="",quantile="1"} Nan
+```
+
+#### Serialize using custom interface
+By default, we shall implement JSON and Prometheus Data Model serialization of metrics. We shall expose a new Serialization interface that users can implement to customize serialization.
+Pros: In-house and no external dependencies.
+Cons: It could be challenging to keep it in sync with Prometheus data model versions.
+      Adding support for histogrma quantiles is not simple.
+
+#### Serializing using Prometheus-CPP:
+Pros: Popular and simplifies histogram and summary metric maintenance.
+      The library has implemented unit tests and integration tests for serialization and data correctness.
+Cons: External dependency.
+
+Both of these approaches are prototyped in [this PR](https://github.com/prestodb/presto/pull/21599/files#).
+
```

Comment:
Maybe we should also consider having a configurable allowlist of metrics, typically histogram till the exporter is not in a sidecar